src/dune/cram_exec.ml

```diff
@@ -1,6 +1,99 @@
 open Stdune
-open Cmdliner
-open Import.Let_syntax
+
+module Sanitizer : sig
+  [@@@ocaml.warning "-32"]
+
+  module Command : sig
+    type t =
+      { output : string
+      ; build_path_prefix_map : string
+      }
+  end
+
+  val impl_sanitizer :
+    (Command.t -> string) -> in_channel -> out_channel -> unit
+
+  val run_sanitizer :
+       ?temp_dir:Path.t
+    -> prog:Path.t
+    -> argv:string list
+    -> Command.t list
+    -> string list
+end = struct
+  module Command = struct
+    type t =
+      { output : string
+      ; build_path_prefix_map : string
+      }
+
+    let of_sexp (csexp : Sexp.t) : t =
+      match csexp with
+      | List [ Atom build_path_prefix_map; Atom output ] ->
+        { build_path_prefix_map; output }
+      | _ -> Code_error.raise "Command.of_csexp: invalid csexp" []
+
+    let to_sexp { output; build_path_prefix_map } : Sexp.t =
+      List [ Atom build_path_prefix_map; Atom output ]
+  end
+
+  let run_sanitizer ?temp_dir ~prog ~argv commands =
+    let prog = Path.to_string prog in
+    let temp_dir =
+      match temp_dir with
+      | Some d -> d
+      | None -> Temp.create Dir ~prefix:"sanitizer" ~suffix:"unspecified"
+    in
+    let argv = prog :: argv in
+    let fname = Path.relative temp_dir in
+    let stdout_path = fname "sanitizer.stdout" in
+    let stdout =
+      Unix.openfile
+        (Path.to_string stdout_path)
+        [ Unix.O_WRONLY; O_CREAT; O_TRUNC; O_SHARE_DELETE ]
+        0o777
+    in
+    let stdin =
+      let path = fname "sanitizer.stdin" in
+      let csexp = List.map commands ~f:Command.to_sexp in
+      Io.with_file_out ~binary:true path ~f:(fun oc ->
+          List.iter csexp ~f:(Csexp.to_channel oc));
+      Unix.openfile (Path.to_string path)
+        [ Unix.O_RDONLY; O_SHARE_DELETE ]
+        0o666
+    in
+    let pid = Spawn.spawn ~prog ~argv ~stdin ~stdout () in
+    Unix.close stdout;
+    Unix.close stdin;
+    match Unix.waitpid [] (Pid.to_int pid) with
```

Comment:
This will block the execution of Dune completely, we should go through the `Process` module here.

Reply:
Just to clarify, this code is currently unused (the warning is suppressed in the module signature). I'm just keeping this code around in case we ever decide to revive sanitizers. Perhaps we should just delete it and stick to version control?

I changed it to `Process.run` anyhow.